source/code/go/src/plugins/out_oms.go

```diff
@@ -19,6 +20,15 @@ func FLBPluginRegister(ctx unsafe.Pointer) int {
 func FLBPluginInit(ctx unsafe.Pointer) int {
 	Log("Initializing out_oms go plugin for fluentbit")
 	InitializePlugin(ContainerLogPluginConfFilePath)
+	enableTelemetry := output.FLBPluginConfigKey(ctx, "EnableTelemetry")
+	telemetryPushInterval := output.FLBPluginConfigKey(ctx, "TelemetryPushInterval")
+	agentVersion := output.FLBPluginConfigKey(ctx, "AgentVersion")
+
+	if strings.Compare(strings.ToLower(enableTelemetry), "true") == 0 {
+		go SendContainerLogFlushRateMetric(telemetryPushInterval, agentVersion)
```

Comment:
why not call telemetry initialize from pluginInit (here) rather than do it in Send**Metric ?

Reply:
there is a dependency on TelemetryClient. I have moved it to the same goroutine.